tools/geom.go

```diff
@@ -0,0 +1,450 @@
+	for sc.Scan() {
+		line := sc.Text()
+		if strings.HasPrefix(line, nDataPairsLine) {
+			nPairs, err := strconv.Atoi(rightofEquals(line))
+			if err != nil {
+				return pairs, err
+			}
+			pairs, err = dataPairsfromTextBlock(sc, nPairs, colWidth, valueWidth)
+			if err != nil {
+				return pairs, err
+			}
+			break
+		}
+	}
+	return pairs, nil
+}
+
+// distance returns the distance along a straight line in euclidean space
+func distance(p0, p1 [2]float64) float64 {
+	result := math.Sqrt(math.Pow((p1[0]-p0[0]), 2) + math.Pow((p1[1]-p0[1]), 2))
+	return result
+}
+
+// pointAtDistance returns a new point along a straight line in euclidean space
+// at a specified distance
+func pointAtDistance(p0, p1 [2]float64, delta float64) [2]float64 {
+	distanceRatio := delta / distance(p0, p1)
+	newX := (1-distanceRatio)*p0[0] + distanceRatio*p1[0]
+	newY := (1-distanceRatio)*p0[1] + distanceRatio*p1[1]
+	return [2]float64{newX, newY}
+}
+
+// interpZ creates a new point a given distance along a line composed
+// of many segments.
+func interpXY(xyPairs [][2]float64, d float64) [2]float64 {
+	// newPoint is an x, y pair
+	var newPoint [2]float64
+	lineSegments := len(xyPairs) - 1
+	lineLength := 0.0
+
+findLineSegment:
+	for i := 0; i < lineSegments; i++ {
+		p0, p1 := xyPairs[i], xyPairs[i+1]
+		lineLength += distance(p0, p1)
+
+		switch {
+		case lineLength > d:
+			delta := distance(p0, p1) - (lineLength - d)
+			newPoint = pointAtDistance(p0, p1, delta)
+			break findLineSegment
+
+		default:
+			continue
+		}
+
+	}
+
+	return newPoint
+}
+
+// attributeZ using station from cross-section line and gis coordinates
+// Minor bug here caused (I think) where interpolating
+// a station beyond the length of the cutline occurs resulting
+// in a point (0,0). Need to add improved error handling for this.
+// For now don't add stations/elevations longer than the GIS line
+func attributeZ(xyPairs [][2]float64, mzPairs [][2]float64) []xyzPoint {
+	points := []xyzPoint{}
+	startingStation := mzPairs[0][0]
+	for _, mzPair := range mzPairs {
+		newPoint := interpXY(xyPairs, mzPair[0]-startingStation)
+		if newPoint[0] != 0 && newPoint[1] != 0 {
+			points = append(points, xyzPoint{newPoint[0], newPoint[1], mzPair[1]})
+		}
+	}
+	return points
+}
+
+func getTransform(sourceProjection string, destinationEPSG int) (gdal.CoordinateTransform, error) {
+	transform := gdal.CoordinateTransform{}
+	sourceSpRef := gdal.CreateSpatialReference(sourceProjection)
+	sourceSpRef.MorphFromESRI()
+	if err := sourceSpRef.Validate(); err != nil {
+		return transform, errors.New("Unable to extract geospatial data. Invalid source Projection")
+	}
+
+	destinationSpRef := gdal.CreateSpatialReference("")
+	if err := destinationSpRef.FromEPSG(destinationEPSG); err != nil {
+		return transform, err
+	}
+	transform = gdal.CreateCoordinateTransform(sourceSpRef, destinationSpRef)
+	return transform, nil
+}
+
+func flipXYLineString(xyLineString gdal.Geometry) gdal.Geometry {
```

Comment:
These flip XY functions will be deprecated once that change Randy was looking at is added to the gdal library, right?

Reply:
Yes, these are just temporary.